logstash-core/lib/logstash/output_delegator_strategy_registry.rb

```diff
@@ -0,0 +1,28 @@
+module LogStash; class OutputDelegatorStrategyRegistry
+  # This is generally used as a singleton
+  # Except perhaps during testing
+  def self.instance
+    @instance ||= self.new
```

Comment:
No change required, but ruby has a `Singleton` library in stdlib that does this.